rllib/agents/ppo/ddppo.py

```diff
@@ -125,15 +125,9 @@ def validate_config(self, config):
 
         # Auto-train_batch_size: Calculate from rollout len and
         # envs-per-worker.
-        if config["train_batch_size"] == -1:
-            config["train_batch_size"] = (
-                config["rollout_fragment_length"] * config["num_envs_per_worker"]
-            )
-        # Users should not define `train_batch_size` directly (always -1).
-        else:
-            raise ValueError(
-                "Set rollout_fragment_length instead of train_batch_size " "for DDPPO."
-            )
+        config["train_batch_size"] = (
```

Comment:
personally, I'd rather we don't modify config dict inside validate_config.
we should simply check if trian_batch_size is set, and raise Value error here.
where we actually need train_batch_size, we will just compute it from rollout_fragment_length * num_envs_per_worker.
personally, I feel like that's a cleaner fix.

what do you guys all think?

Reply:
I'm in agreement with you @gjoliver. What do you think @dsctt?

Reply:
I looked more into the issue, and I think the cleanest change you could do is the following:

override `__init__` of the ddppo trainer (still call `super().__init__()`). 
After calling the super constructor, do the modification on `config["rollout_fragment_length"]`. We could throw a warning or an error in the constructor if a user has already specified this in the config dict already. 


The reason why I suggest this over your approach @gjoliver is because this config parameter is only used in underlying modules such as `MultiGPUTrainOneStep` and `TrainOneStep`. It would probably be annoying/ to override these functions/not be as clean in implementation.

Reply:
Sounds good to me. Could use a pointer as to where we should now set train_batch_size. Is it in execution_plan() of the same file or do we need to override another function?

Reply:
makes sense. as long as it's after validate_config(), we should be good.

Reply:
By the modification, do you mean set `train_batch_size` using `rollout_fragment_length` * `num_envs_per_worker` like before? If that works, then I think a warning is good if it wasn't =-1, telling the user that we are changing it. @avnishn

Reply:
hmmm -- yes set `train_batch_size`. But let's still raise an error here instead of a warning (but inside of the constructor). Users need to know that their parameters aren't supposed to be set, and would be overwritten. Sometimes these warnings aren't that easy to catch in the console log, unless its being logged on every single training/logging iteration.

Reply:
Looks like we won't always have access to the full config in `__init__` since the config gets merged with the default_config in `setup()` [here](https://github.com/ray-project/ray/blob/master/rllib/agents/trainer.py#L841). So, in the case that the `train_batch_size` is not specified by the user, there is a key error. What do you think?

Reply:
I suppose with a little more code we could check right there in `__init__` if the config has the keys and then use the default values directly.

Reply:
This is how I went about it. Let me know if that may cause issues.